src/azure-cli-core/azure/cli/core/cli_subprocess.py

```diff
@@ -0,0 +1,108 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=no-else-return
+
+import subprocess
+import re
+from knack.log import get_logger
+
+
+__all__ = [
+    # recommended invoking function
+    "run",
+    # advanced use cases
+    "CliPopen",
+    # older high-level API, can be replaced by run in many cases, support for existing codes
+    "call",
+    "check_call",
+    "check_output",
+]
+
+logger = get_logger(__name__)
+
+ARG_MASK_CHAR = re.compile(r'[$;\s]')
+
+
+def subprocess_arg_type_check(args, kwargs):
+    skip_arg_type_check = kwargs.get("skip_arg_type_check", None)
+    if skip_arg_type_check is not None:
+        del kwargs["skip_arg_type_check"]
+    if skip_arg_type_check:
+        return
+    if isinstance(args, list):
+        return
+    from azure.cli.core.azclierror import ArgumentUsageError
+    raise ArgumentUsageError("cli_subprocess args should be a list of sequence")
+
+
+def subprocess_arg_mask(args, kwargs):
+    enable_arg_mask = kwargs.get("enable_arg_mask", None)
```

Comment:
`enable_arg_mask` and `skip_arg_type_check` are hidden in `kwargs`. In that case, it should be included in the doc string. We can't expect developers to look into the source code to know them.